packages/dappmanager/src/modules/https-portal/migration.ts

```diff
@@ -138,3 +144,36 @@ export function migrateCoreNetworkAndAliasInCompose(
 
   compose.write();
 }
+
+function isComposeNetworkAndAliasMigrated(
+  composeNetwork: ComposeNetwork | null,
+  serviceNetwork: ComposeServiceNetwork | null,
+  composeVersion: string,
+  alias: string
+): boolean {
+  // semver can only work with version of 3 numbers. e.g: 3.5.0
+  const composeVersionCleaned = semver.clean(composeVersion + ".0", {
+    loose: true
+  });
+  const minimumComposeVersionCleaned = semver.clean(
+    params.MINIMUM_COMPOSE_VERSION + ".0",
+    {
+      loose: true
+    }
+  );
+
+  if (!composeVersionCleaned || !minimumComposeVersionCleaned)
+    throw Error("Docker compose file version cannot be used by semver");
```

Comment:
Is it really necessary to throw an error here? Would that prevent the migration of other packages? If you really want to throw, consider adding a try / catch in the for loop iterating over packages

Reply:
semver.clean may throw a null version. I guess it is almost impossible taking into account the docker compose file version format (2 digits and I add a 0 at the end), but just in case

Reply:
Then isolate that into `parseComposeVersion()` with signature
```ts
function parseComposeVersion(v: string): string
```
and it throws if clean return falsy